andreeapetcu/assignment2/BinaryTreeNode.java

```diff
@@ -0,0 +1,81 @@
+public class BinaryTreeNode<T> {
+  public T value;
```

Comment:
You do an equality test (.equals()) on this in your Ancestors class. But there is nothing protecting it from being null. I would be tempted to put a check or an assert in the constructor (often called a precondition check) failing if a null value is used. That way you won't get a strange null pointer exception later.

I'd also recommend using an accessor class rather than a public field. This is Item 14 in "Effective java" 2nd edition.